drivers/timer/cortex_m_systick.c

```diff
@@ -202,7 +203,13 @@ void z_clock_set_timeout(s32_t ticks, bool idle)
 		delay =
```

Comment:
delay = CYC_PER_TICK * ceiling_fraction(delay, CYC_PER_TICK);

would be more succinct and clear, perhaps.